float64 carrotDistance
float64 carrotDirection
